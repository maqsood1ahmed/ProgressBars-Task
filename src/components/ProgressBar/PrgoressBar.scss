$textColor: rgb(109, 255, 51);

.progress-bar {
  color: $textColor;
}