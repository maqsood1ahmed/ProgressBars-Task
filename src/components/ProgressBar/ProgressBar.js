import React from 'react';

import propTypes from 'prop-types';

const progressBar = {
    width: '100%',
    height: '50px',
    boxShadow: '0px 0px 0px 1px rgba(0,0,0,0.66)'
};

const ProgressBar = props => {
    let percentageLimit = props.percentageLimit(0, props.percentage, 100) + '%';
    const progressBarProgress = {
        width: percentageLimit,
        height: '100%',
        backgroundColor: props.percentage > 100 ? 'red' : '#3498db',
        marginTop: '10px',
        position: 'relative',
        transition: 'width 0.8s'
    };
    let textStyle = {
        position: 'absolute',
        marginTop: '-45px',
        left: '49%',
        fontSize: '25px'
    };
    return (
        <div className="progress-bar" style={progressBar}>
            <div
                className="progress-bar-progress"
                style={progressBarProgress}
            />
            <p style={textStyle}>{props.percentage}%</p>
        </div>
    );
};

ProgressBar.propTypes = {
    percentage: propTypes.number,
    percentageLimit: propTypes.func
};

export default ProgressBar;
