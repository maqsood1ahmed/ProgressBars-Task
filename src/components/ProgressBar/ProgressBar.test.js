import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProgressBar from './ProgressBar';

configure({ adapter: new Adapter() });

describe('ProgressBar test', () => {
    it('should be defined', () => {
        expect(ProgressBar).toBeDefined();
    });
    it('should render correctly', () => {
        const baseProps = {
            percentage: 40,
            percentageLimit: jest.fn()
        };
        const tree = shallow(<ProgressBar {...baseProps} />);
        expect(tree).toMatchSnapshot();
    });
});
