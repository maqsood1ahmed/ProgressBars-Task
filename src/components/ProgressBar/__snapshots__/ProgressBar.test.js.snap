// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar test should render correctly 1`] = `ShallowWrapper {}`;
